pkg/command/update/update.go

```diff
@@ -0,0 +1,236 @@
+  kubectl rw update example-rw -n foo -mr 256Mi
+
+  # Update memory request of meta component in risingwave named example-rw in namespace foo and group test.
+  kubectl rw update example-rw -n foo -c meta -g test -mr 256Mi
+`
+)
+
+type Options struct {
+	*cmdcontext.BasicOptions
+
+	computeRequest Request
+	computeLimit   Request
+	memoryRequest  Request
+	memoryLimit    Request
+	group          string
+	component      string
+}
+
+// request holds the request string and converted resource quantity.
+type Request struct {
+	requested_qty string
+	converted_qty k8sresource.Quantity
+}
+
+func NewCommand(ctx *cmdcontext.RWContext, streams genericclioptions.IOStreams) *cobra.Command {
+	o := Options{
+		BasicOptions: cmdcontext.NewBasicOptions(streams),
+	}
+
+	cmd := &cobra.Command{
+		Use:     "update",
+		Short:   "Update CPU and memory configuration for risingwave instances",
+		Long:    LongDesc,
+		Example: Example,
+		Run: func(cmd *cobra.Command, args []string) {
+			util.CheckErr(o.Complete(ctx, cmd, args))
+			util.ExitOnErr(o.Validate(ctx, cmd, args))
+			util.CheckErr(o.Run(ctx, cmd, args))
+		},
+	}
+
+	cmd.Flags().StringVar(&o.computeRequest.requested_qty, "cpurequest", "", "The target cpu request.")
+	cmd.Flags().StringVar(&o.computeLimit.requested_qty, "cpulimit", "", "The target cpu limit.")
+	cmd.Flags().StringVar(&o.memoryRequest.requested_qty, "memoryrequest", "", "The target memory request.")
+	cmd.Flags().StringVar(&o.memoryLimit.requested_qty, "memorylimit", "", "The target memory limit.")
+	cmd.Flags().StringVarP(&o.group, "group", "g", util.DEFAULT_GROUP, "The group to be updated. If not set, update the default group.")
+	cmd.Flags().StringVarP(&o.component, "component", "c", util.GLOBAL, "The component to be updated. If not set, update global resources.")
+
+	return cmd
+}
+
+func (o *Options) Validate(ctx *cmdcontext.RWContext, cmd *cobra.Command, args []string) error {
+	// parse the resource requests
+	if o.computeRequest.requested_qty != "" {
+		qty, err := k8sresource.ParseQuantity(o.computeRequest.requested_qty)
+		if err != nil {
+			return err
+		}
+		o.computeRequest.converted_qty = qty
+	}
+
+	if o.computeLimit.requested_qty != "" {
+		qty, err := k8sresource.ParseQuantity(o.computeLimit.requested_qty)
+		if err != nil {
+			return err
+		}
+		o.computeLimit.converted_qty = qty
+	}
+
+	if o.memoryRequest.requested_qty != "" {
+		qty, err := k8sresource.ParseQuantity(o.memoryRequest.requested_qty)
+		if err != nil {
+			return err
+		}
+		o.memoryRequest.converted_qty = qty
+	}
+
+	if o.memoryLimit.requested_qty != "" {
+		qty, err := k8sresource.ParseQuantity(o.memoryLimit.requested_qty)
+		if err != nil {
+			return err
+		}
+		o.memoryLimit.converted_qty = qty
+	}
+
+	// validate group
+	if o.group == "" {
+		o.group = util.DEFAULT_GROUP
+	} else {
+		rw, err := o.GetRwInstance(ctx)
+		if err != nil {
+			return err
+		}
+
+		if !util.IsValidGroup(rw, o.group) {
```

Comment:
IIUC, it's component-specific? So we can also pass the component to the validate function.

Reply:
Sorry, could you elaborate more on this?

Reply:
> Sorry, could you elaborate more on this?

I mean, it should be like `IsValidGroup(component, group`.